### DISCUSSION TITLE
MikroOrm + NestJS + QueryBuilder

### DISCUSSION THREAD

**Discussion by lmnogues** [mikro-orm/mikro-orm#6303] [General]
### Describe the bug

Hi,

We are using MikroOrm with NestJs.
We have a usecase where the request need to go throught a QueryBuilder.
On run the Querybuilder is failing with a message asking for the client password.
I've tracked the issue and it seems that when the entityManager create the QueryBuilder it's only using the 'default' configuration from the mikro-orm.config.ts file and not it's full configuration.

We have 2 files to configure our connection :
the root mikro-orm.config.ts with some default param : 
```
/** Common MikroOrm options. */
const DEFAULT_OPTIONS: Options = {
  extensions: [Migrator],
  // forceUndefined changes not supported by other modules
  // forceUndefined: true,
  entities: Entities.ENTITIES,
  driver: PostgreSqlDriver,
  metadataProvider: TsMorphMetadataProvider,
  metadataCache: {
    options: { cacheDir: './dist/cache' },
  },
  schema: 'bridge',
  debug: false,
};

/** MikroOrm options factory: merge provided options with default conf. */
export function mikroOrmConfBuilder(opts: Options = {}): Options {
  return { ...DEFAULT_OPTIONS, ...opts };
}
```

the nest databaseprovider file with a `ForRootAsync` which allow us to use a configuration Service (for param validation) :

```
useFactory: (config: DatabaseConfigService) => {
        return mikroOrmConfBuilder({
          clientUrl: config.url,
          pool: {
            min: 0,
            max: Number(config.poolsize),
          },
          debug: config.logging,
          replicas: config.read_replicas_urls.map((replica_url) => ({
            clientUrl: replica_url,
          })),
          schema: config.schema,
          migrations: {
            migrationsList,
            tableName: 'mikro_orm_migrations',
            snapshot: false,
          },
        });
      },
```

The issue is reproduced on the `this.em.getConnection()` as well.

And on the `this.em.execute` the issue is here as well except if we have the `@Transactional` decorator

### Reproduction

the following is not working in our setup, because part of the config is lost
```
const connection = this.em.getConnection('read');
connection.ecute('....');

const qb = this.em.createQueryBuilder('entity');
[...]
qb.getResults()
```

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.4.1

### Node.js version

node 22.12.0 / typescript 5.7.2

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lmnogues** [comment]
apparently wrapping the querybuilder avoid the issue 
```
  await this.em.transactional(async (em) => {
      const qb = em.createQueryBuilder(Entity);
      qb.select('id');
      const result = await qb.getResultList();
      console.log(result);
    });
```

**Comment by B4nan** [comment]
It's not entirely clear what you are trying to report here, but it's most likely a misunderstanding of how things work behind the scenes. In your reproduction example, the query using `connection.execute()` will use a different connection handle than the QB itself. The only way to run the queries on the same connection handle is to actually run things inside a transaction.

**Reply by lmnogues** [reply]
the fact that it's a different connection is not an issue, the issue is that the new connection is not built with the same parameter as the connection from the entityManager, and is loosing half it's config

**Reply by B4nan** [reply]
There is one knex instance under the hood for the whole ORM, all connections are handed by knex, so I doubt it would be the case. Hard to guess what exactly you mean by that without seeing a complete reproduction.

**Reply by lmnogues** [reply]
I'll try to provide a full repo

**Comment by lmnogues** [comment]
The issue seems to not be ONLY about connection;

I've put the repo here : https://github.com/lmnogues/mikro-orm_nestjs_pnp_issue
the issue seems to depends if we use `pnp`or `node-module`  in yarn.

if you try to execute my project it will break asking for some password.
if you change `yarnrc.yaml`file on line 5 to `nodeLinker: node-modules` it will work immediatly...
